src/operations/find.ts

```diff
@@ -15,16 +15,19 @@ import { Sort, formatSort } from '../sort';
 import { isSharded } from '../cmap/wire_protocol/shared';
 import { ReadConcern } from '../read_concern';
 import type { ClientSession } from '../sessions';
-import type { Projection } from '../mongo_types';
 
-/** @public */
-export interface FindOptions<TSchema = Document> extends CommandOperationOptions {
+/**
+ * @public
+ * @typeParam TSchema - Unused schema definition, deprecated usage, only specify `FindOptions` with no generic
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export interface FindOptions<TSchema extends Document = Document> extends CommandOperationOptions {
   /** Sets the limit of documents returned in the query. */
   limit?: number;
   /** Set to sort the documents coming back from the query. Array of indexes, `[['a', 1]]` etc. */
   sort?: Sort;
   /** The fields to return in the query. Object of fields to either include or exclude (one of, not both), `{'a':1, 'b': 1}` **or** `{'a': 0, 'b': 0}` */
-  projection?: Projection<TSchema>;
+  projection?: Document;
```

Comment:
Wouldn't setting this to `TSchema` make the default functionality the same but still allow a more granular level of control without breaking things?

Reply:
You mean setting `projection?: TSchema`? I think that would cause issue with the typical projection of inclusion / exclusion like `{ _id: 0, name: 1 }`. where `name: string` in the TSchema